gateway-server/src/main/java/org/apache/knox/gateway/services/token/impl/AliasBasedTokenStateService.java

```diff
@@ -92,9 +93,17 @@ protected long getMaxLifetime(final String token) {
   @Override
   public long getTokenExpiration(final String token) throws UnknownTokenException {
     long expiration = 0;
-
-    validateToken(token);
-
+    try {
+      validateToken(token);
+    } catch (final UnknownTokenException e) {
+      /* if token permissiveness is enabled we check JWT token expiration when the token state is unknown */
+      if (permissiveFailureEnabled && StringUtils
+          .containsIgnoreCase(e.toString(), "Unknown token")) {
```

Comment:
What? We catch an `UnknownTokenException` and then are doing a string check on the error message? Why?

Reply:
Because this is how the validate() function let's you know that the TokenStateService does not have the token, by throwing UnknownTokenException.

Reply:
To Kevin's point, why do you need to check the message if it's an UnknownTokenException?

Reply:
I see the confusion now, I started working on it before "UnknownTokenException" was introduced so we have "IllegalArgumentException" this is an artifact of that. I will change it.